### PR TITLE
ENG 1752 - slash command auto completion

### PR DESCRIPTION
**Pull request by dexhorthy**
## What problem(s) was I solving?

Implemented slash command autocomplete functionality to improve command discovery and usability in the WUI. Previously, users had to remember exact command names with no assistance or visibility into available commands, and couldn't distinguish between repository-specific and user-wide commands.

Related issues:
- [ENG-1752](https://linear.app/humanlayer/issue/ENG-1752/slash-command-auto-completion): slash-command auto-completion

### Slash Command Autocomplete
- Type `/` at the start of a message to see all available commands
- Commands discovered from both local (`.claude/commands/`) and global (`~/.claude/commands/`) directories
- Visual source indicators show "local" or "global" badges next to each command
- Fuzzy search filters commands as you type (e.g., `/plan` matches both `/create_plan` and `/implement_plan`)
- Keyboard navigation with arrow keys and Tab/Shift+Tab
- Press Enter to select a command, Space to close dropdown
- Commands from subdirectories use colon notation (e.g., `/hl:research`)
- Commands appear as mention nodes in the editor before submission
- Seamless conversion to plain text when message is sent
- Global commands take precedence over local ones when duplicates exist

### Backend API
- New `/api/v1/slash-commands` endpoint for command discovery
- Real-time filesystem reading from both local and global command directories
- Returns source field ('local' or 'global') for each command
- Automatic deduplication with global commands taking precedence
- Support for nested command directories with colon convention
- Fuzzy search filtering on the backend for consistent results

### Backend Implementation (`hld/`)
- Added `GetSlashCommands` handler in `sessions.go` that reads from filesystem
- Walks both `.claude/commands/` (local) and `~/.claude/commands/` (global) directories recursively
- Uses map-based deduplication where global commands override local ones with same name
- Converts subdirectories to colon notation (e.g., `hl/research.md` → `/hl:research`)
- Implements fuzzy search using `github.com/sahilm/fuzzy` library
- Returns each command with source field indicating 'local' or 'global' origin
- Returns empty array (not null) when no commands exist
- Added comprehensive unit and integration tests with filesystem mocking

### Frontend Integration (`humanlayer-wui/`)
- Created `SlashCommandList.tsx` component based on existing `FileMentionList` pattern
- Extended Tiptap editor with new Mention instance for slash commands
- Configured to trigger only at start of line with `/` character
- Added keyboard navigation (arrows, tab, enter, escape)
- Integrated with daemon client to fetch commands from backend
- Fixed double-slash bug in draft launcher by using JSON processing
- Updated TypeScript types to include source field in SlashCommand interface
- Added muted text badges showing "local" or "global" for each command
- Styled with `text-muted-foreground` and `bg-muted` for subtle appearance

### API Changes
- Added OpenAPI specification for `/slash-commands` endpoint with source field
- Generated TypeScript SDK with updated `getSlashCommands` method
- Updated all generated API models and types to include source information

## How to verify it

- [x] I have ensured `make check test` passes

### Manual Testing
1. Start a session in WUI
2. Create test commands:
   - Add a local command: `.claude/commands/test_local.md`
   - Add a global command: `~/.claude/commands/test_global.md`
   - Add a duplicate: same name in both directories
3. Type `/` at the beginning of the message input:
   - Verify dropdown appears with available commands
   - Check that commands show source badges ("local" or "global")
   - Verify duplicate commands show only once (global version)
4. Type additional characters after `/`:
   - Verify fuzzy search filters the list
   - Try `/plan` - should show `/create_plan` and `/implement_plan`
   - Try `/test` - should show both test commands with correct sources
5. Use keyboard navigation:
   - Arrow keys move selection up/down
   - Tab/Shift+Tab cycle through commands
   - Enter selects and inserts command
   - Space closes dropdown
   - Escape closes dropdown
6. Submit a message with a slash command:
   - Verify command appears as plain text in the message
7. Test with subdirectories (if any exist):
   - Verify colon notation works (e.g., `/hl:research`)
8. Test edge cases:
   - Empty `.claude/commands/` directory shows "No commands found"
   - Non-existent directory handled gracefully
   - Commands with same name in both directories show global version only

## Description for the changelog

Added slash command autocomplete in WUI with fuzzy search, keyboard navigation, and dual-source command discovery from both local (`.claude/commands/`) and global (`~/.claude/commands/`) directories. Commands display source indicators and support nested directories with colon notation. Global commands take precedence when duplicates exist.

### CODE REVIEW

File: humanlayer-wui/src/lib/daemon/http-client.ts [hunk 1]
```diff
@@ -123,6 +123,17 @@ export class HTTPDaemonClient implements IDaemonClient {
...
 
+  async getSlashCommands(params: {
+    sessionId: string
+    query?: string
+  }): Promise<{ data: Array<{ name: string; source: 'local' | 'global' }> }> {
+    await this.ensureConnected()
+
+    const response = await this.client!.getSlashCommands(params)
+
+    return response as { data: Array<{ name: string; source: 'local' | 'global' }> }
```
Comment (balanceiskey) [inline humanlayer-wui/src/lib/daemon/http-client.ts:134]: minor, but a thing to watch out for, there should be return type available for this

File: humanlayer-wui/src/components/internal/SessionDetail/SessionDetail.tsx [hunk 2]
```diff
@@ -611,8 +611,29 @@ function SessionDetail({ session, onClose }: SessionDetailProps) {
...
       setIsLaunchingDraft(true)
 
-      // Get the prompt text from the editor
-      const prompt = responseEditor?.getText() || ''
+      // Get the editor content and process mentions/slash commands properly
+      let prompt = ''
+      if (responseEditor) {
+        const json = responseEditor.getJSON()
+
+        const processNode = (node: any): string => {
```
[the remark this answers is not in the document]
Comment (balanceiskey) [reply humanlayer-wui/src/components/internal/SessionDetail/SessionDetail.tsx:619]: This seems valid

File: humanlayer-wui/src/components/internal/SessionDetail/SessionDetail.tsx [hunk 3]
```diff
@@ -611,8 +611,29 @@ function SessionDetail({ session, onClose }: SessionDetailProps) {
     try {
       setIsLaunchingDraft(true)
 
-      // Get the prompt text from the editor
-      const prompt = responseEditor?.getText() || ''
```
Comment (balanceiskey) [inline humanlayer-wui/src/components/internal/SessionDetail/SessionDetail.tsx:615]: Making an assumption here that `getText()` was not working as expected once we introduced slash commands?
